app/com/gu/floodgate/reindex/BulkJobActor.scala

```diff
@@ -0,0 +1,220 @@
+package com.gu.floodgate.reindex
+
+import akka.actor.{ Actor, ActorLogging, Props }
+import akka.pattern.pipe
+import com.gu.floodgate.{ CustomError, RunningJobNotFound }
+import com.gu.floodgate.contentsource.{ ContentSource, ContentSourceSettings }
+import com.gu.floodgate.jobhistory.JobHistoryService
+import com.gu.floodgate.reindex.BulkJobActor._
+import com.gu.floodgate.runningjob.{ RunningJob, RunningJobService }
+import com.gu.scanamo.DynamoFormat
+import com.typesafe.scalalogging.StrictLogging
+
+import scala.util.{ Failure, Success }
+import org.joda.time.{ DateTime, DateTimeZone }
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+object BulkJobActor {
+
+  def props(contentSources: List[ContentSource], runningJobService: RunningJobService, reindexService: ReindexService, jobHistoryService: JobHistoryService) =
+    Props(new BulkJobActor(contentSources, runningJobService, reindexService, jobHistoryService))
+
+  sealed trait BulkReindexRequestResult
+  sealed trait DropPendingJobResult
+  case object CanTrigger extends BulkReindexRequestResult
+  case object RejectJob extends BulkReindexRequestResult
+
+  case object StartBulkReindex
+  case object ReindexNext
+  case object IsActorReindexing
+
+  case class ReindexStarted(id: String, environment: String)
+  case class ReindexError(id: String, environment: String, error: CustomError)
+  case class Polling(id: String, environment: String)
+
+  case class RunningJobInfo(name: String, id: String, env: String, documentsIndexed: Int, documentsExpected: Int, startTime: DateTime, settings: ContentSourceSettings)
+  case class PendingJobInfo(name: String, id: String, env: String)
+  case class CompletedJobInfo(name: String, id: String, env: String, startTime: DateTime, finishTime: DateTime, status: ReindexStatus)
+
+  case class IsReindexing(runningJobs: Seq[RunningJobInfo], pendingJobs: List[PendingJobInfo], completedJobs: List[CompletedJobInfo]) extends BulkReindexRequestResult
+  case object NotReindexing extends BulkReindexRequestResult
```

Comment:
Can we group these a bit more clearly?
E.g. each trait with its associated types, and keep incoming message types together